File: 02_recursion_basics/02_06_reverse_array/reverse_array.js
const reverse_array = (start, end, arr) => {
    if (start < end) {
        [arr[start], arr[end]] = [arr[end], arr[start]];
        reverse_array(start + 1, end - 1, arr);
    }
}

const array = [1, 2, 3, 4, 5, 6, 7, 8, 9]
console.log("Before:-");
console.log(array);
reverse_array(0, array.length - 1, array);
console.log();
console.log("After:-");
console.log(array);

File: 02_recursion_basics/02_06_reverse_array/reverse_array.py
def reverseArray(start, end, arr):
    if start < end:
        arr[start], arr[end] = arr[end], arr[start]
        reverseArray(start + 1, end - 1, arr)
        
items = [1, 2, 3, 4, 5, 6, 7, 8, 9]
print("Before:-")
print(items)
reverseArray(0, len(items) - 1, items)
print("After:-")
print(items)